From 3dbb81eec537a8b4f353186c72fdd96b9c625904 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Fri, 1 Oct 2021 23:28:46 -0400
Subject: [PATCH] automatically load deepmd

---
 src/lammps.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lammps.cpp b/src/lammps.cpp
index 89c59bec3d..2fe9bef1f5 100644
--- a/src/lammps.cpp
+++ b/src/lammps.cpp
@@ -861,6 +861,7 @@ void LAMMPS::post_create()
       if (strcmp(suffix2,"omp") == 0) input->one("package omp 0");
     }
   }
+  input->one("plugin load libdeepmd_lmp.so");
 
   // invoke any command-line package commands
 
-- 
2.31.1

