From a3a84dc06d6b17ed728e30aa709eabffea377cd6 Mon Sep 17 00:00:00 2001
From: Jinzhe Zeng <jinzhe.zeng@rutgers.edu>
Date: Wed, 4 Aug 2021 18:48:26 -0400
Subject: [PATCH] low precesion

---
 cmake/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 42e6d12ffb..ece5e908d1 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -77,7 +77,7 @@ set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "Use compiler extensions")
 ########################################################################
 # User input options                                                   #
 ########################################################################
-set(LAMMPS_MACHINE "" CACHE STRING "Suffix to append to lmp binary (WON'T enable any features automatically")
+set(LAMMPS_MACHINE "low" CACHE STRING "Suffix to append to lmp binary (WON'T enable any features automatically")
 mark_as_advanced(LAMMPS_MACHINE)
 if(LAMMPS_MACHINE)
   set(LAMMPS_MACHINE "_${LAMMPS_MACHINE}")
@@ -661,8 +661,8 @@ if(BUILD_SHARED_LIBS)
     add_custom_target(
       install-python
       ${Python_EXECUTABLE} install.py -v ${LAMMPS_SOURCE_DIR}/version.h
-      -m ${LAMMPS_PYTHON_DIR}/lammps.py
-      -l ${CMAKE_BINARY_DIR}/liblammps${CMAKE_SHARED_LIBRARY_SUFFIX}
+	  -m ${LAMMPS_PYTHON_DIR}/lammps${LAMMPS_MACHINE}.py
+	  -l ${CMAKE_BINARY_DIR}/liblammps${LAMMPS_MACHINE}${CMAKE_SHARED_LIBRARY_SUFFIX}
       WORKING_DIRECTORY  ${LAMMPS_PYTHON_DIR}
       COMMENT "Installing LAMMPS Python module")
   else()
@@ -695,7 +695,7 @@ if(BUILD_SHARED_LIBS OR PKG_PYTHON)
       -c "import distutils.sysconfig as cg; print(cg.get_python_lib(1,0,prefix='${CMAKE_INSTALL_PREFIX}'))"
       OUTPUT_VARIABLE PYTHON_DEFAULT_INSTDIR OUTPUT_STRIP_TRAILING_WHITESPACE)
     set(PYTHON_INSTDIR ${PYTHON_DEFAULT_INSTDIR} CACHE PATH "Installation folder for LAMMPS Python module")
-    install(FILES ${LAMMPS_PYTHON_DIR}/lammps.py DESTINATION ${PYTHON_INSTDIR})
+	install(FILES ${LAMMPS_PYTHON_DIR}/lammps${LAMMPS_MACHINE}.py DESTINATION ${PYTHON_INSTDIR})
   endif()
 endif()
 
-- 
2.31.1

